src/Proto3/Suite/DotProto/Generate.hs

```diff
@@ -685,12 +755,45 @@ dotProtoEnumD parentIdent enumIdent enumParts =
          predFailure     = match_ (HsIdent "pred") [ HsPWildCard ]
                                   (HsUnGuardedRhs (HsApp predErrorE enumNameE)) []
 
+         parseJSONPBDecls :: [HsMatch]
+         parseJSONPBDecls =
+           [ let pat nm =
```

Comment:
Large list comprehensions like this one, while not _difficult_ to parse, certainly take a minute to wrap my head around. Perhaps do notation in the list monad would make this more readable?

Reply:
I think it's only marginally more readable via use of the list monad, but then it sticks out stylistically from the rest of the module which makes regular use of list comprehensions. /shrug. Going to leave it as-is, but point taken.